packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/platform_interface/platform_interface_collection_reference.dart

```diff
@@ -48,4 +48,16 @@ abstract class CollectionReferencePlatform extends QueryPlatform {
   DocumentReferencePlatform doc([String path]) {
     throw UnimplementedError("doc() is not implemented");
   }
+
+  @override
+  bool operator ==(dynamic o) =>
+      o is CollectionReferencePlatform &&
+      o.firestore == firestore &&
+      o._pointer == _pointer;
```

Comment:
I think checking the path is better here, since it removes a dependency on the Pointer class and matches a property local to the current class.